model/billing_record.rb

```diff
@@ -14,20 +14,14 @@ def active
   include ResourceMethods
 
   def duration(begin_time, end_time)
-    # We bill some records based on the amount, others on duration. For
-    # 'VmCores' billing records, the core counts are stored in the amount column
-    # and charges are based on duration. For example, 10 minutes of standard-4
-    # vm usage would be calculated as `10 (duration) x 2 (amount) x rate_for_standard_core`.
-    # 'GitHubRunnerMinutes' billing records, on the other hand, store used
-    # minutes in the amount column and charges are based on this amount. For
-    # instance, 10 minutes of standard-4 runner usage would be calculated as
-    # `1 (duration) x 10 (amount) x rate_for_standard_4_runner_minute`.
-    # 'GitHubRunnerMinutes' is the only category we currently bill by amount,
-    # though network costs could potentially be another suitable candidate. At
-    # present, we only perform this check for 'GitHubRunnerMinutes'. While a
-    # more generic solution may be beneficial, we are waiting to gather more
-    # data points before undertaking a more significant refactor.
-    return 1 if billing_rate["resource_type"] == "GitHubRunnerMinutes"
+    # Billing logic differs based on the resource type: some are billed by duration, others
+    # by amount. For 'VmCores' billing records, the core counts are stored in the amount
+    # column and charges are based on duration. For example, 10 minutes of standard-4 vm
+    # usage would be calculated as `10 (duration) x 2 (amount) x rate_for_standard_core`.
+    # 'GitHubRunnerMinutes' and 'InferenceTokens' records store the used minutes or tokens
+    # in the 'amount' column, and billing is based on that amount.
+    # For records billed by amount, the duration is always set to 1.
+    return 1 if billing_rate["billed_by"] == "amount"
```

Comment:
I like this `amount` vs `minutes` improvement. 🚀 cc @byucesoy Since he knows our billing system well too and we discussed it before.

Reply:
It could be a separate commit with more detailed commit message, though: introduce `billing_by` column, and introduce inference tokens billing rates.

Reply:
Split into separate commits.